pandas/core/indexes/interval.py

```diff
@@ -440,23 +440,26 @@ def is_monotonic(self):
         Return True if the IntervalIndex is monotonic increasing (only equal or
         increasing values), else False
         """
-        return self._multiindex.is_monotonic
+        return self.is_monotonic_increasing
 
     @cache_readonly
     def is_monotonic_increasing(self):
         """
         Return True if the IntervalIndex is monotonic increasing (only equal or
         increasing values), else False
         """
-        return self._multiindex.is_monotonic_increasing
+        values = [self.right, self.left]
+
+        sort_order = np.lexsort(values)
+        return Index(sort_order).is_monotonic
```

Comment:
Hey @makbigc,

Thansk for contributing.

I worked on ``CategoricalIndex.is_monotonic_increasing`` a while ago and I can remember that instantiating an Index was quite heavy, probably because a lot of things are being set up when instantiation an Index. Can you try using ``pd._libs.algos.is_monotonic(sort_order, False)`` instead of making an index. Should be even faster.

I get from a quick timeit test:

```python
>>> N = 10**5
>>> left = np.append(np.arange(N), np.array(0))
>>> right = np.append(np.arange(1, N + 1), np.array(1))
>>> sort_order = np.lexsort([left, right])
>>> %timeit  pd.Index(sort_order).is_monotonic
42.8 µs ± 3.48 µs per loop
>>> %timeit pd._libs.algos.is_monotonic(sort_order, False)[0]
4.15 µs ± 123 ns per loop
```
So a 10x improvement.

Architectually, can you also see if it makes sense to add ``InterValTree.is_monotonic_[increasing|decreasing]`` methods using ``_libs.algos.is_monotonic`` and then in ``IntervalIndex.is_monotonic_increasing`` just make a call to ``self._engine.is_monotonic_increasing``. That would make the call structure for IntervalIndex similar to the one for other indexes, see e.g. ``Index.is_monotonic_increasing``.

 @jreback, do you agree?

Reply:
@topper-123 Thanks for your elucidative advice. The use of algos.is_monotonic does improve the performance.

```
All benchmarks:

       before           after         ratio
     [23650ef5]       [3e32db5b]
     <GH24813-inc-dec~2>       <GH24813-inc-dec>
          213±3ns          210±2ns     0.98  index_object.IntervalIndexMethod.time_monotonic_inc(1000)
          268±1ns          260±2ns     0.97  index_object.IntervalIndexMethod.time_monotonic_inc(100000)
```

While putting `is_monotonic_increasing` in `IntervalIndex._engine` slows down the performance. I guess this is due to the formation of `IntervalTree` is expensive in the `_engine` method. The order is increased from ns to ms.

Reply:
You're not really measuring the speed of the operation in the above benchmark. Notice that ``@cache_readonly`` above the ``is_monotonic_increasing`` method. That means that the result of the call to ``is_monotonic_increasing`` replaces the method itelf upon first call, so you're just measuring attribute access speed.

The formation of IntervalTree should happen only once, so should be in the ns-range also on subsequent calls. Are you forgtting to cache the result of ``IntervalTree.is_monotonic_increasing``? If it resly is being constructed anew each time, can you post an issue to the issue list.